server/plugin/api.go

```diff
@@ -329,13 +329,47 @@ func (p *Plugin) handleGetSubscriptions(w http.ResponseWriter, r *http.Request)
 	offset, limit := p.GetOffsetAndLimitFromQueryParams(r)
 
 	channelID := r.URL.Query().Get(constants.QueryParamChannelID)
+	serviceType := r.URL.Query().Get(constants.QueryParamServiceType)
+	eventType := r.URL.Query().Get(constants.QueryParamEventType)
 	project := r.URL.Query().Get(constants.QueryParamProject)
 	if project != "" {
 		subscriptionByProject := []*serializers.SubscriptionDetails{}
 		for _, subscription := range subscriptionList {
 			if subscription.ProjectName == project {
 				if channelID == "" || subscription.ChannelID == channelID {
-					subscriptionByProject = append(subscriptionByProject, subscription)
+					switch serviceType {
+					case "", constants.FilterAll:
+						subscriptionByProject = append(subscriptionByProject, subscription)
+					case constants.FilterBoards:
+						switch eventType {
+						case "", constants.FilterAll:
+							if subscription.EventType == constants.SubscriptionEventWorkItemCreated ||
+								subscription.EventType == constants.SubscriptionEventWorkItemUpdated ||
+								subscription.EventType == constants.SubscriptionEventWorkItemDeleted ||
+								subscription.EventType == constants.SubscriptionEventWorkItemCommented {
+								subscriptionByProject = append(subscriptionByProject, subscription)
+							}
+						default:
+							if subscription.EventType == eventType {
+								subscriptionByProject = append(subscriptionByProject, subscription)
+							}
+						}
+					case constants.FilterRepos:
+						switch eventType {
+						case "", constants.FilterAll:
+							if subscription.EventType == constants.SubscriptionEventPullRequestCreated ||
+								subscription.EventType == constants.SubscriptionEventPullRequestMerged ||
+								subscription.EventType == constants.SubscriptionEventPullRequestUpdated ||
+								subscription.EventType == constants.SubscriptionEventPullRequestCommented ||
+								subscription.EventType == constants.SubscriptionEventCodePushed {
```

Comment:
I think you should create something like this and use that here
```
var ValidSubscriptionEventsForRepos = map[string]bool {
    SubscriptionEventPullRequestCreated: true,
    SubscriptionEventPullRequestMerged: true,
}
```
You should create two maps for boards and repos in the constants package and then use those maps here.